### PR TITLE
Closes #18253: Bookmark and History open new tabs in the background

### LINKED ISSUE #18253: [Feature request] Bookmark opened in new tab should open in background
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
Go to bookmark menu, browse any deep folder level bookmark, click 3 dot menu and select open in New tab
It opens in current bookmark tab only increasing frustration of a user who might wants to open another bookmark as the menu and folder path is gone. User has to open bookmark again, browse to the folder and select a new one... Painful process 

### Expected behavior
On clicking open in New tab it should actually open in New tab in the background without overriding current bookmark view

### Actual behavior
When clicking open in tab it is simply overriding current view and opening in current tab only 

### Device information

* Device vendor / model and Android version: ? A10
* Firefox for Android version: ? (go to Settings -> About Firefox): latest nightly

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by rocketsroger** [comment]
https://user-images.githubusercontent.com/6579541/116318289-8aba8a00-a782-11eb-8f7a-9f6e196501f8.mp4

**Review by topotropic — COMMENTED** [review]
Thanks! I know @vesta0  answered to the original issue so it might be good for her to review it! I don't think we need to open the tab tray when people open a tab in the background but just show a snackbar – similar to what we do when people open links in the background when on a site.

**Comment by rocketsroger** [comment]
> Thanks! I know @vesta0 answered to the original issue so it might be good for her to review it! I don't think we need to open the tab tray when people open a tab in the background but just show a snackbar – similar to what we do when people open links in the background when on a site.

@topotropic Thanks for reviewing.   It's my mistake that I didn't check with you first.  I was trying to behave the same as when we open multiple bookmark or history tabs.  Current tabs tray is shown when opening multiple tabs, shouldn't we behave the same when just opening one bookmark or history tab in the background?

**Comment by topotropic** [comment]
@rocketsroger If people just open one bookmark via the menu, we can show the snackbar ("New tab opened" | "Switch", "New private tab opened" | "Switch") that we currently use when people open a link via context menu on a site. Let me know if you have any questions.  Thank you!

**Comment by rocketsroger** [comment]
> @rocketsroger If people just open one bookmark via the menu, we can show the snackbar ("New tab opened" | "Switch", "New private tab opened" | "Switch") that we currently use when people open a link via context menu on a site. Let me know if you have any questions. Thank you!

Sounds good.  I'll update my implementation.  Thanks!

**Comment by rocketsroger** [comment]
The snack bar implementation will be much bigger then originally estimated.  As discussed offline, I've created an issue https://github.com/mozilla-mobile/fenix/issues/19365 to track the follow up work.  Thanks,

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt [hunk 1]
```diff
@@ -169,7 +173,7 @@ class DefaultBookmarkController(
         }
     }
 
-    private fun openInNewTab(
+    private fun openInNewTabAndShow(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt:176]: nice

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 2]
```diff
@@ -263,6 +258,21 @@ class HistoryFragment : LibraryPageFragment<HistoryItem>(), UserInteractionHandl
...
     }
 
+    private fun openItemAndShowTray(item: HistoryItem, mode: BrowsingMode) {
+        when (mode.isPrivate) {
+            true -> requireComponents.analytics.metrics.track(Event.HistoryOpenedInPrivateTab)
+            false -> requireComponents.analytics.metrics.track(Event.HistoryOpenedInNewTab)
+        }
+
+        (activity as HomeActivity).browsingModeManager.mode = mode
+        (activity as HomeActivity).components.useCases.tabsUseCases.let { tabsUseCases ->
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:268]: nit
```suggestion
        val homeActivity = activity as HomeActivity
        homeActivity.browsingModeManager.mode = mode
        homeActivity.components.useCases.tabsUseCases.let { tabsUseCases ->
```

**Review by jonalmeida — APPROVED** [review]
LGTM ✅

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt (hunk 2 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt]: I'll update.  Thanks!

**Pull request merged**